core/src/main/java/io/questdb/cutlass/pgwire/CleartextPasswordPgWireAuthenticator.java

```diff
@@ -0,0 +1,512 @@
+                processStartupMessage(msgLen);
+                break;
+            case INIT_CANCEL_REQUEST:
+                processCancelMessage();
+                return Authenticator.NEEDS_DISCONNECT;
+            case INIT_SSL_REQUEST:
+                compactRecvBuf();
+                prepareSslResponse();
+                state = State.WRITE_AND_EXPECT_INIT_MESSAGE;
+                break;
+            case INIT_GSS_REQUEST:
+                compactRecvBuf();
+                prepareGssResponse();
+                state = State.WRITE_AND_EXPECT_INIT_MESSAGE;
+                break;
+            default:
+                LOG.error().$("unknown init message [protocol=").$(protocol).$(']').$();
+                throw BadProtocolException.INSTANCE;
+        }
+        return Authenticator.OK;
+    }
+
+    private int processPasswordMessage() throws BadProtocolException {
+        int availableToRead = availableToRead();
+        if (availableToRead < 1 + Integer.BYTES) { // msgType + msgLen
+            return Authenticator.NEEDS_READ;
+        }
+        byte msgType = Unsafe.getUnsafe().getByte(recvBufReadPos);
+        assert msgType == MESSAGE_TYPE_PASSWORD_MESSAGE;
+
+        int msgLen = getIntUnsafe(recvBufReadPos + 1);
+        long msgLimit = (recvBufReadPos + msgLen + 1); // +1 for the type byte which is not included in msgLen
+        if (recvBufWritePos < msgLimit) {
+            return Authenticator.NEEDS_READ;
+        }
+
+        // at this point we have a full message available ready to be processed
+        recvBufReadPos += 1 + Integer.BYTES; // first move beyond the msgType and msgLen
+
+        long hi = PGConnectionContext.getStringLength(recvBufReadPos, msgLimit, "bad password length");
+        dbcs.of(recvBufReadPos, hi);
+        if (userDatabase.match(username, dbcs)) {
+            recvBufReadPos = msgLimit;
+            compactRecvBuf();
+            prepareLoginOk();
+            state = State.WRITE_AND_AUTH_SUCCESS;
+        } else {
+            LOG.error().$("bad password for user [user=").$(username).$(']').$();
+            prepareWrongUsernamePasswordResponse();
+            state = State.WRITE_AND_AUTH_FAILURE;
+        }
+        return Authenticator.OK;
+    }
+
+    private void processStartupMessage(int msgLen) throws BadProtocolException {
+        long msgLimit = (recvBufStart + msgLen);
+        long lo = recvBufReadPos;
+
+        // there is an extra byte at the end, and it has to be 0
+        while (lo < msgLimit - 1) {
+            final long nameLo = lo;
+            final long nameHi = PGConnectionContext.getStringLength(lo, msgLimit, "malformed property name");
+            final long valueLo = nameHi + 1;
+            final long valueHi = PGConnectionContext.getStringLength(valueLo, msgLimit, "malformed property value");
+            lo = valueHi + 1;
+
+            // store user
+            dbcs.of(nameLo, nameHi);
+            if (Chars.equals(dbcs, "user")) {
```

Comment:
nit: have a look at `CairoKeywords` there is a faster method to check for ascii constant